.circleci/config.yml

```
@@ -29,10 +29,14 @@ jobs:
           export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD)
           echo "$WEBAPP_GIT_COMMIT"
 
-          trap 'export WEBAPP_GIT_COMMIT=$(git rev-parse HEAD^); echo "previous $WEBAPP_GIT_COMMIT"; curl -f -o ./dist.tar.gz https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1;' ERR
-          curl -f -o ./dist.tar.gz https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz
-          mkdir ./dist && tar -xvf ./dist.tar.gz -C ./dist --strip-components=1
-          trap - ERR
+          FILE_DIST=dist.tar.gz
+          until curl --max-time 5 -f -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; do echo waiting for webapp build; sleep 110; done;
```

Why do we need to sleep for 110 seconds?

---

https://app.circleci.com/pipelines/github/mattermost/mattermost-webapp/42646/workflows/8e2227e4-bb96-4d9f-bfc5-64e84e321450

I wanted to check 5 times during a full cycle until it is uploaded. But turns out it would be 361seconds until s3 upload. So ~370/5=~75seconds . 
Or nope? Just sleep 5?

---

I am wondering if we could use a timeout approach instead (e.g. https://jeromebelleman.gitlab.io/posts/devops/until/) and attempt multiple times without waiting for  such a long time. We can still add an initial sleep if we know it won't be ready for a while but at least we get to poll and fail/succeed faster afterwards.

---

OK. What about
```suggestion
          until curl --max-time 30 -f -o $FILE_DIST https://pr-builds.mattermost.com/mattermost-webapp/commit/${WEBAPP_GIT_COMMIT}/mattermost-webapp.tar.gz; do echo waiting for webapp build; sleep 5; done;
```
for now?

---

Sure :+1: